File: ui/src/models/courseSchedule.ts
import { Grade, PrimarySchoolSubject, SecondarySchoolSubject } from "./postings"

export interface PrimarySchoolCourse {
    id: string,
    subject: PrimarySchoolSubject,
    grades: Grade[],
    startTime: string, // only use the time in these 
    endTime: string
    location?: string | null;
}

export interface SecondarySchoolCourse {
    id: string,
    subject: SecondarySchoolSubject,
    grades: Grade[],
    startTime: string, // only use the time in these 
    endTime: string
    location?: string | null;
}

File: api/api/Models/CourseSchedule.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models;

public class PrimarySchoolCourse
{
    public string Id { get; set; }
    [ForeignKey("AspNetUsers")]
    public User user {  get; set; } 

    public List<Grade> grades {  get; set; }
    public PrimarySchoolSubject subject;

    public DateTime startTime;
    public DateTime endTime;

    public string? location;
}

public class SecondarySchoolCourse
{
    public string Id { get; set; }
    [ForeignKey("AspNetUsers")]
    public User user { get; set; }

    public List<Grade> grades { get; set; }
    public SecondarySchoolSubject subject;

    public DateTime startTime;
    public DateTime endTime;

    public string? location;
}

